nameko/retry.py

```diff
@@ -0,0 +1,53 @@
+import functools
+import itertools
+from time import sleep
+
+import wrapt
+
+
+class RetryDelay(object):
+    def __init__(self, delay, backoff, max_delay):
+        self.delay = delay
```

Comment:
Yeah, I lifted it from our application codebase. Actually it's previously been part of the Nameko codebase too, but we reverted it in favour of a less generic solution (which has since been rolled back).

I think if it's going to live as a helper function it's useful to keep the `max_delay` argument.